### PR TITLE
Remove dead parse_enum, Type::from_str, and 4 unused deps

### PR DESCRIPTION
**Pull request by DominicBurkart**
## Summary

- Removes the `parse_enum` function (~90 lines) and `Type::from_str` implementation (~55 lines) from `nodes.rs` -- both are dead code superseded by `pest_ast_builder.rs` (`build_enum_decl`, `build_field_type`)
- Removes 4 dependencies from `marigold-grammar/Cargo.toml` that were only used by the dead code or not used at all: `regex`, `lazy_static`, `rusymbols`, `once_cell`
- Removes `#[macro_use] extern crate lazy_static` from `lib.rs`
- Extracts the duplicated `runner_code()` method (identical in `StreamVariableNode` and `StreamVariableFromPriorStreamVariableNode`) into a shared helper function

Net reduction: ~147 lines of code, 4 compile-time dependencies.

### Why this is safe

- `parse_enum` is never called anywhere in the codebase -- enum parsing is handled entirely by `PestAstBuilder::build_enum_decl`
- `Type::from_str` is only called inside `parse_enum` (for recursive Option type parsing) and nowhere else -- type parsing is handled by `PestAstBuilder::build_field_type`
- `regex` is only used inside the removed dead code
- `lazy_static` is only used inside the removed dead code
- `rusymbols` has zero usages anywhere in the codebase
- `once_cell` is only referenced in generated code strings (runtime dep of `marigold-impl`, not a compile dep of `marigold-grammar`)

## Test plan

- [ ] `cargo check -p marigold-grammar` compiles successfully
- [ ] `cargo test -p marigold-grammar` -- all existing tests pass (none tested the dead code)
- [ ] `cargo check -p marigold-grammar --features io` compiles with io feature
- [ ] Full workspace `cargo test` passes

### CODE REVIEW

**Review by DominicBurkart — COMMENTED** [review]
**(Draft)** Good cleanup -- removing `parse_enum`, `Type::from_str`, and the 4 unused deps is well-motivated and the deduplication of `runner_code()` is a nice bonus.

**Blocker:** All 11 CI checks are failing due to a typo in `Cargo.toml` -- `itetools` instead of `itertools`. See inline comment. Once fixed, this should be green.

The rest of the diff looks correct: `regex`, `lazy_static`, `rusymbols`, and `once_cell` removals are safe since they were only used by the dead code being removed.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by DominicBurkart — COMMENTED** [review]
**(Draft -- post-fix review)** The `itetools` typo from the previous review has been fixed and all 11 CI checks now pass on the head commit (`9fa749f`). Re-reviewing the full diff:

**Verdict: LGTM -- no issues found.**

Summary of changes (3 files, +11 / -168):

1. **`Cargo.toml`** -- Removes `regex`, `lazy_static`, `rusymbols`, `once_cell`. All four are confirmed unused after the dead code removal. Remaining deps are untouched.
2. **`lib.rs`** -- Removes `#[macro_use] extern crate lazy_static;`. Correct since the crate is gone.
3. **`nodes.rs`** -- Removes `parse_enum` (~91 lines) and `Type::from_str` (~55 lines), both dead code superseded by `pest_ast_builder.rs`. Extracts the duplicated `runner_code()` method into a shared free function. The extracted function is identical to both originals -- no behavioral change.

**No bugs, no secrets/PII, no missing tests** (the removed code had no test coverage, and the `runner_code` extraction is a pure refactor covered by existing integration tests). No new code paths introduced.

**Suggestion for when marking ready-for-review:** The PR description's test plan checkboxes can all be checked off now that CI is green. Consider squashing the 3 commits (initial cleanup, split commit, typo fix) into a single commit before merging for a cleaner history.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by DominicBurkart — COMMENTED** [review]
**Review of head SHA `988ee749` — LGTM (would approve)**

All 11 CI checks pass. No prior review exists against this commit. Full diff reviewed below.

---

### Summary of changes (3 files, +11 / -168)

**`marigold-grammar/Cargo.toml`** — Removes `regex`, `lazy_static`, `rusymbols`, `once_cell`.
- `regex` and `lazy_static`: only used inside the removed `parse_enum` and `Type::from_str` dead code. Safe.
- `rusymbols`: zero usages anywhere in the codebase. Safe.
- `once_cell`: correctly identified as a runtime dep of `marigold-impl` (referenced in generated code strings), not a compile-time dep of `marigold-grammar`. Safe to drop here.
- `itertools = "0.10.2"` is present and correctly spelled — the `itetools` typo from an earlier commit is resolved.

**`marigold-grammar/src/lib.rs`** — Removes `#[macro_use] extern crate lazy_static;`. Correct since the crate is no longer a dependency.

**`marigold-grammar/src/nodes.rs`** — Four logical changes:
1. Removes `use regex::Regex` and `use std::str::FromStr` imports. Both only served the deleted code; no orphaned usages remain in the file.
2. Removes `impl FromStr for Type` (~55 lines) — dead code; type parsing is handled entirely by `PestAstBuilder::build_field_type`. No external callsites confirmed.
3. Removes `pub fn parse_enum` (~91 lines) — dead code; enum parsing is handled entirely by `PestAstBuilder::build_enum_decl`. No external callsites confirmed.
4. Extracts the duplicated `runner_code()` method from `StreamVariableNode` and `StreamVariableFromPriorStreamVariableNode` into a private free function `fn runner_code(variable_name: &str) -> String`. The extracted body is byte-for-byte identical to both originals — no behavioral change. Correctly scoped as `fn` (not `pub fn`), with a doc comment explaining the sharing context.

---

### Checks

| Area | Finding |
|---|---|
| Bugs | None. The `runner_code` extraction is a pure refactor; the `format!` template is unchanged. |
| Architecture | Sound. Removing the regex-based parsing path in favor of the pest-AST path (`PestAstBuilder`) is the correct long-term direction. The private free function pattern for `runner_code` is idiomatic. |
| Missing tests | None needed. The removed code had no test coverage. The `runner_code` extraction is exercised by existing integration tests that cover `StreamVariableNode`/`StreamVariableFromPriorStreamVariableNode` code generation. The `Benches` CI job passes. |
| Bloat | Net -157 lines of Rust source, -4 compile-time dependencies. Binary size and incremental compile time improve. |
| PII / secrets | None present. |

---

### Minor note (non-blocking)

The unresolved inline review thread on `Cargo.toml` (the `itetools` typo comment from an earlier review cycle) is now outdated — the fix landed in this commit. Resolving that thread before merge will keep the review view clean.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: marigold-grammar/Cargo.toml [hunk 1]
```diff
@@ -14,14 +14,10 @@ tokio = []
 async-std = []
 
 [dependencies]
-regex = "1"
 num-traits = "0.2"
```
[the remark this answers is not in the document]
Comment (DominicBurkart) [reply marigold-grammar/Cargo.toml:17]: Fixed in commit `9fa749f` ("fix: correct typo itetools → itertools in Cargo.toml"). The current branch head has `itertools = "0.10.2"` correctly spelled and all 11 CI checks pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
